### PR TITLE
[ct] add lsr reboot to reboot_random_validators

### PR DESCRIPTION
**Pull request by davidiw**
* Add the ability to randomly remove lsr instances to the reboot random validators
* Fixed lsr instance mapping in cluster_builder
* Added structured logging support for lsr

### CODE REVIEW

**Comment by davidiw** [comment]
test fails due to lsr trying to re-initialize storage, which is purely a config-builder issue:


kubernetes.pod_name:lsr-13 @timestamp:Jul 15, 2020 @ 22:15:32.717 log:thread 'main' panicked at 'Unable to initialize backend storage: Key already exists: consensus', consensus/safety-rules/src/persistent_safety_storage.rs:56:9 stream:stderr time:Jul 15, 2020 @ 22:15:32.717 kubernetes.namespace_name:default kubernetes.pod_id:013102ce-ace4-4ed9-92b2-af542b1a549e kubernetes.labels.app:libra-lsr kubernetes.labels.libra-node:true kubernetes.labels.peer_id:lsr-13 kubernetes.annotations.kubernetes_io/psp:eks.privileged kubernetes.host:ip-192-168-198-57.us-west-2.compute.internal kubernetes.container_name:main kubernetes.docker_id:2858dcc59b89fe3371a24e2257a46d564645317308a9ffcd7acf3942f46585a2 kubernetes.container_hash:013141ce43c2b76e412b923bd2f6c5bae66351c6f252b8ce50a604c8b4fc8fa5 _id:3rEMVnMBEYRArVInpEPN _type:flb_type _index:kubernetes_cluster-2020.07.16 _score: -
--




we can chat tomorrow about how to resolve this, but this has nothing to do with the code I've written here, which should be good for review.

File: testsuite/cluster-test/src/cluster_builder.rs [hunk 1]
```diff
@@ -143,7 +143,7 @@ impl ClusterBuilder {
                         )
                     })
                     .collect();
-                lsrs.append(&mut vault_instances);
+                vaults.append(&mut vault_instances);
```
Comment (andll) [inline testsuite/cluster-test/src/cluster_builder.rs:146]: :)

File: testsuite/cluster-test/src/experiments/reboot_random_validator.rs [hunk 2]
```diff
@@ -37,13 +52,32 @@ impl ExperimentParam for RebootRandomValidatorsParams {
...
             );
         }
-        let mut instances = Vec::with_capacity(self.count);
-        let mut all_instances = cluster.validator_instances().to_vec();
-        let mut rnd = rand::thread_rng();
-        for _i in 0..self.count {
-            let instance = all_instances.remove(rnd.gen_range(0, all_instances.len()));
-            instances.push(instance);
+
+        if self.lsr_count > cluster.validator_instances().len() {
```
[the remark this answers is not in the document]
Comment (davidiw) [reply testsuite/cluster-test/src/experiments/reboot_random_validator.rs:56]: yeah, this is my bad, copypasta :/

**Pull request closed without merging**